zebra-scan/src/service/tests.rs

```diff
@@ -77,3 +77,51 @@ pub async fn scan_service_deletes_keys_correctly() -> Result<()> {
 
     Ok(())
 }
+
+/// Tests that results are cleared are deleted correctly
+#[tokio::test]
+pub async fn scan_service_clears_results_correctly() -> Result<()> {
+    let mut db = new_test_storage(Network::Mainnet);
+
+    let zec_pages_sapling_efvk = ZECPAGES_SAPLING_VIEWING_KEY.to_string();
+
+    for fake_result_height in [Height::MIN, Height(1), Height::MAX] {
+        db.insert_sapling_results(
+            &zec_pages_sapling_efvk,
+            fake_result_height,
+            fake_sapling_results([
+                TransactionIndex::MIN,
+                TransactionIndex::from_index(40),
+                TransactionIndex::MAX,
+            ]),
+        );
+    }
+
+    assert!(
+        !db.sapling_results(&zec_pages_sapling_efvk).is_empty(),
+        "there should be some results for this key in the db"
+    );
+
+    let (mut scan_service, _cmd_receiver) = ScanService::new_with_mock_scanner(db.clone());
+
+    let response = scan_service
+        .ready()
+        .await
+        .map_err(|err| eyre!(err))?
+        .call(Request::ClearResults(vec![zec_pages_sapling_efvk.clone()]))
+        .await
+        .map_err(|err| eyre!(err))?;
+
+    match response {
+        Response::ClearedResults => {}
+        _ => panic!("scan service returned unexpected response variant"),
+    };
+
+    assert_eq!(
+        db.sapling_results(&zec_pages_sapling_efvk).len(),
+        1,
+        "all results for this key should have been deleted, one empty entry should remain"
+    );
```

Comment:
```suggestion
    );
    
    // TODO: Make sure the result is empty. 
```

Reply:
Added a check to the test in https://github.com/ZcashFoundation/zebra/pull/8219/commits/70e176e748f415cb6d977b7e09c93fd802a5b6f3